src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,150 @@
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import frc.robot.utils.SparkMaxUtils;
+
+public class Shooter extends SubsystemBase{
+
+  private final CANSparkMax motorA =
+      new CANSparkMax(RobotMap.SHOOTER_MOTOR_A_CAN_ID, MotorType.kBrushless);
+  private final CANSparkMax motorB =
+      new CANSparkMax(RobotMap.SHOOTER_MOTOR_B_CAN_ID, MotorType.kBrushless);
+  private final CANSparkMax pivotMotor =
+      new CANSparkMax(RobotMap.SHOOTER_PIVOT_MOTOR_CAN_ID, MotorType.kBrushless);
+  private final AbsoluteEncoder pivotMotorAbsoluteEncoder =
+      pivotMotor.getAbsoluteEncoder(Type.kDutyCycle);
+  private double pivotDesiredPosition = ShooterCal.STARTING_POSITION_DEGREES;
+
+  private double prevVelocity = 0;
+  private Optional<Double> prevTimestamp = Optional.empty();
+
+  /** double distance (meters), double angle (degrees) for pivotMotor */
+  private InterpolatingDoubleTreeMap pivotAngleMap;
+
+  private final SparkPIDController controllerA = motorA.getPIDController();
+  private final SparkPIDController controllerB = motorB.getPIDController();
+  private final ProfiledPIDController pivotController = new ProfiledPIDController(ShooterCal.PIVOT_MOTOR_kP, ShooterCal.PIVOT_MOTOR_kI, ShooterCal.PIVOT_MOTOR_kD, new TrapezoidProfile.Constraints(ShooterCal.PIVOT_MAX_VELOCITY_DEG_PER_SECOND, ShooterCal.PIVOT_MAX_ACCELERATION_DEG_PER_SECOND_SQUARED));
+
+  public Shooter() {
+    SparkMaxUtils.initWithRetry(this::initSparks, Constants.SPARK_INIT_RETRY_ATTEMPTS);
+    pivotAngleMap = new InterpolatingDoubleTreeMap();
+    pivotAngleMap.put(Constants.PLACEHOLDER_DOUBLE, Constants.PLACEHOLDER_DOUBLE);
+    pivotAngleMap.put(Constants.PLACEHOLDER_DOUBLE, Constants.PLACEHOLDER_DOUBLE);
+  }
+
+  public void shoot() {
+    controllerA.setReference(ShooterConstants.SHOOTER_MOTOR_SPEED_RPM, ControlType.kVelocity);
+    controllerB.setReference(ShooterConstants.SHOOTER_MOTOR_SPEED_RPM, ControlType.kVelocity);
+  }
+
+  public void stop() {
+    controllerA.setReference(0, ControlType.kVelocity);
+    controllerB.setReference(0, ControlType.kVelocity);
+  }
+
+  /** Returns the cosine of the intake angle in degrees off of the horizontal. */
+  public double getCosineArmAngle() {
+    return Math.cos(
+        getPivotPosition() - ShooterConstants.POSITION_WHEN_HORIZONTAL_DEGREES);
+  }
+
+  public void controlPosition(double distance){
+    pivotDesiredPosition = pivotAngleMap.get(distance);
+    pivotController.setGoal(pivotDesiredPosition);
+    double timestamp = Timer.getFPGATimestamp();
+
+    double armDemandVoltsA = pivotController.calculate(getPivotPosition());
+    double armDemandVoltsB = pivotController.calculate(getPivotPosition());
+    if (!prevTimestamp.isEmpty()) {
+      armDemandVoltsB += ShooterCal.PIVOT_MOTOR_FF.calculate(prevVelocity, pivotController.getSetpoint().velocity, timestamp - prevTimestamp.get());
+    }
```

Comment:
Maybe `else` we do the `calculate` with just the `final double nextVelocityRpm = pivotController.getSetpoint().velocity`?